src/renderers/webgl/WebGLProgram.js

```diff
@@ -700,7 +701,7 @@ function WebGLProgram( renderer, cacheKey, parameters, bindingStates ) {
 	vertexShader = unrollLoops( vertexShader );
 	fragmentShader = unrollLoops( fragmentShader );
 
-	if ( parameters.isWebGL2 && parameters.isRawShaderMaterial !== true ) {
```

Comment:
This is probably the most controversial part of my change. Feedback welcome!

Reply:
I would be good to find a solution without changing this line since it affects user code.

Just to be clear, you have removed `parameters.isRawShaderMaterial` to make the extension work, right?

Reply:
Would it be possible to use `ShaderMaterial` in `PMREMGenerator` instead?

Reply:
Yes, that's right. I also considered using `ShaderMaterial` instead; it would be a bigger change and I can't currently remember why I chose raw in the first place, so I'm afraid I might hit a blocker. Still, doesn't it seem odd that we insert the `customExtensions` into the `RawShaderMaterial`, but then don't fix them up for WebGL2? This is why it felt like a bug to me.

Reply:
Aren't these extensions automatically enabled with WebGL 2? So you need the injection only with WebGL 1.

Reply:
well, I only removed the check for `RawShaderMaterial`, so this is only changing the behavior for RawShaderMaterials.  the code this adds translates from the old function names like `texture2DGradEXT` to the WebGL2 version `texture`. That's very convenient; is there a better way?

Reply:
Hmm... I'll merge and do another PR reverting some of these things so we can see how it looks.

Reply:
IMO, it's best when `PMREMGenerator` uses `ShaderMaterial`. All changes to the renderer can then be reverted.

The idea of excluding `RawShaderMaterial` from the GLSL 3.0 conversion process was to give the user full control over the GLSL. I don't think we need to change this policy for fixing `PMREMGenerator`.

Reply:
Gotcha. I didn't realize a `RawShaderMaterial` couldn't be used to support both WebGL1 and WebGL2. I guess `ShaderMaterial` is the way to go then. Might be good to add that to the docs somewhere.